edk2toollib/uefi/edk2/path_utilities_test.py

```diff
@@ -761,3 +761,44 @@ def test_get_absolute_path_on_this_system_from_edk2_relative_path(self):
 
         # pass in bad parameter
         self.assertIsNone(pathobj.GetAbsolutePathOnThisSystemFromEdk2RelativePath(None))
+    
+    def test_get_absolute_path_then_relative_path_when_path_contains_repeated_packagepath_name(self):
+        ''' Test the back and forth between GetAbsolutePath and GetRelativeFromAbsolute when the
+        path structure has multiple instances of a package path
+        File layout:
+         root/                  <-- current working directory (self.tmp)
+            PlatformClientPkg/           <-- workspace root
+                CLIENT_PKG/       <-- packages path
+                    ClientPkg/   <-- A edk2 package
+                        ClientPkg.DEC
+                        module1/
+                            module1.INF
+                        module2/
+                            module2.INF
+                            x64/
+                               TestFile.c
+        '''
+        ws_rel = "PlatformClientPkg"
+        ws_abs = os.path.join(self.tmp, ws_rel)
+        os.mkdir(ws_abs)
+
+        folder_pp1_rel = "CLIENT_PKG"
+        folder_pp1_abs = os.path.join(ws_abs, folder_pp1_rel)
+        os.mkdir(folder_pp1_abs)
+
+        ws_pkg_name = "ClientPkg"
+        ws_pkg_abs = self._make_edk2_package_helper(folder_pp1_abs, ws_pkg_name)
```

Comment:
I don't see you actually create the ClientPkg or ClientPkg.dec file or the Module1/2 stuff?  Comment should match code and test case.  Am I missing something?

Reply:
Module1 / Module2 is a remanent from copy / pasting from another test. Will remove from the comment as it is not important.

As for creating ClientPkg / ClientPkg.dec, The files don't actually need to exist to be able to determine the relative path. This capability / functionality is used for PR evaluation when PR has a file or directory removed. However, I think it is best that the test still creates ClientPkg and ClientPkg.dec, so I will add that in.